clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -1343,6 +1430,12 @@ CodeGenFunction::emitBuiltinObjectSize(const Expr *E, unsigned Type,
   assert(Ptr->getType()->isPointerTy() &&
          "Non-pointer passed to __builtin_object_size?");
 
+  if (IsDynamic)
```

Comment:
Why did this code block move to here from up there?

Reply:
At this point, we know that `EmittedE` has a value. See line 1429.